Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java

```diff
@@ -198,32 +196,50 @@ void WhenUpdateExamWithNonExistingExamId_ShouldRespondNotFound() throws Exceptio
                 .andExpect(status().isNotFound());
     }
 
+    @DisplayName("Given two groups G1(student: A) and G2(student: B, C), and an exam," +
+            "When add the two groups as examinees, Then the examinees of the exam should be (A,B,C).")
     @Test
-    void WhenAddExaminees_A_B_C_ShouldSucceedAndRespondErrorEmailList_C() throws Exception {
-        var exam = createExamAndGet(new Date(), new Date(), "sample exam");
+    void testAddTwoGroupsOfExamineesFrom() throws Exception {
+        var exam = createExamAndGet(new Date(), new Date(), "Exam");
+        Group G1 = givenGroupWithMembers("G1", STUDENT_A_ID);
+        Group G2 = givenGroupWithMembers("G2", STUDENT_B_ID, STUDENT_C_ID);
+
+        addGroupsOfExaminees(exam, G1, G2);
+
+        var examinees = getExaminees(exam);
+        shouldHaveExaminees(examinees, STUDENT_A_EMAIL, STUDENT_B_EMAIL, STUDENT_C_EMAIL);
+    }
+
+    private void shouldHaveExaminees(List<Student> actualExaminees, String... expectedExamineeEmails) {
+        assertEquals(new HashSet<>(asList(expectedExamineeEmails)), mapToSet(actualExaminees, Student::getEmail));
+    }
+
+    @Test
+    void WhenAddExaminees_A_B_Z_ShouldSucceedAndRespondErrorEmailList_Z() throws Exception {
+        var exam = createExamAndGet(new Date(), new Date(), "Exam");
 
         List<String> errorEmails = getBody(
-                createExaminees(exam.id, STUDENT_A_EMAIL, STUDENT_B_EMAIL, "studentC@example.com")
+                addExaminees(exam.id, STUDENT_A_EMAIL, STUDENT_B_EMAIL, "studentZ@example.com")
                         .andExpect(status().isOk()), new TypeReference<>() {
                 });
 
         assertEquals(1, errorEmails.size());
-        assertEquals("studentC@example.com", errorEmails.get(0));
+        assertEquals("studentZ@example.com", errorEmails.get(0));
```

Comment:
Why not declare local variable String studentCEmail?

Reply:
unnecessary